### PR TITLE
PLUGIN-63. Hubspot Realtime Source

### PR DESCRIPTION
**Pull request by aonischuk**
Source periodically reads data from Hubspot.
First read fetches all data. Subsequent reads fetch only updates.

Links:
https://wiki.cask.co/display/CE/Hubspot+plugins
https://issues.cask.co/browse/PLUGIN-63

### CODE REVIEW

File: src/main/java/io/cdap/plugin/hubspot/source/streaming/HubspotStreamingSource.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  private HubspotStreamingSourceConfig config;
+
+  public HubspotStreamingSource(HubspotStreamingSourceConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    config.validate(collector); // validate when macros are not substituted
```
Comment (mlozbin-cybervisiontech) [inline src/main/java/io/cdap/plugin/hubspot/source/streaming/HubspotStreamingSource.java:49]: missed collector check

File: src/main/java/io/cdap/plugin/hubspot/source/streaming/PullFrequency.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  public String getName() {
+    return name;
+  }
+
+  public Integer getMinutesValue() {
+    return minutesValue;
+  }
+
+  public static PullFrequency fromValue(String name) {
+    return Arrays.stream(PullFrequency.values()).filter(pullFrequency -> pullFrequency.getName().equals(name))
```
Comment (mlozbin-cybervisiontech) [inline src/main/java/io/cdap/plugin/hubspot/source/streaming/PullFrequency.java]: please start each pipeline call on new line

**Review by mlozbin-cybervisiontech — CHANGES_REQUESTED** [review]

**Review by mlozbin-cybervisiontech — APPROVED** [review]

File: src/main/java/io/cdap/plugin/hubspot/source/streaming/HubspotStreamingSource.java (hunk 1 above)
Comment (aonischuk) [reply src/main/java/io/cdap/plugin/hubspot/source/streaming/HubspotStreamingSource.java:49]: Done

File: src/main/java/io/cdap/plugin/hubspot/source/streaming/PullFrequency.java (hunk 2 above)
Comment (aonischuk) [reply src/main/java/io/cdap/plugin/hubspot/source/streaming/PullFrequency.java]: Done

**Pull request closed without merging**